web/src/components/vis/PcaPage/PcaPage.vue

```diff
@@ -76,6 +76,89 @@ export default {
     },
   },
 };
+</script>-->
+<script lang="ts">
+import {
+  defineComponent, computed, ref, watch,
+} from '@vue/composition-api';
+import VisTileLarge from '@/components/vis/VisTileLarge.vue';
+import LayoutGrid from '@/components/LayoutGrid.vue';
+import store from '../../../store';
+import ScorePlot from './ScorePlot.vue';
+import ScreePlot from './ScreePlot.vue';
+import LoadingsPlot from './LoadingsPlot.vue';
+
+export default defineComponent({
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+  },
+  components: {
+    ScorePlot,
+    ScreePlot,
+    LoadingsPlot,
+    VisTileLarge,
+    LayoutGrid,
+  },
+  setup(props) {
+    const pcXval = ref('1');
+    const pcYval = ref('2');
+    const numComponentsVal = ref('10');
+    const pcX = ref(1);
+    const pcY = ref(2);
+    const numComponents = ref(10);
+    const showEllipses = ref(true);
+    const showCrosshairs = ref(true);
+    const showCutoffs = ref(true);
+    const showScree = ref(true);
+    const showScore = ref(true);
+    const showLoadings = ref(true);
```

Comment:
I think this is a situation where I would have considered `reactive` instead of `refs` for brevity in the return function.

```typescript
    const plot  = reactive({
      pcXval: 1,
      pcYval: 2,
      numComponentsVal: 10,
      /* ... */
    });
    /* .... */
    return {
      ...toRefs(plot),
      ready,
    };
```

Reply:
I'd never heard of `toRefs`, that is the better solution.